commit 2da3654aafd41854277da3927be7af21c2be1e71
Author: dann frazier <dannf@hp.com>
Date:   Mon Feb 15 18:30:42 2010 -0700

    [Adjusted to apply to Debian's 2.6.18]
    commit 690e744869f3262855b83b4fb59199cf142765b0
    Author: Dave Jones <davej@redhat.com>
    Date:   Mon Oct 19 19:55:13 2009 -0400
    
        [SCSI] gdth: Prevent negative offsets in ioctl CVE-2009-3080

diff --git a/drivers/scsi/gdth.c b/drivers/scsi/gdth.c
index 7471696..1b79b5c 100644
--- a/drivers/scsi/gdth.c
+++ b/drivers/scsi/gdth.c
@@ -3347,7 +3347,7 @@ static int gdth_read_event(gdth_ha_str *ha, int handle, gdth_evt_str *estr)
         eindex = handle;
     estr->event_source = 0;
 
-    if (eindex >= MAX_EVENTS) {
+    if (eindex < 0 || eindex >= MAX_EVENTS) {
         spin_unlock_irqrestore(&ha->smp_lock, flags);
         return eindex;
     }
